### PR TITLE
Changes required to use newGenfacadeModel

### PR DESCRIPTION
**Pull request by Anipik**
Related PR https://github.com/dotnet/arcade/pull/2221

### CODE REVIEW

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <NoWarn>$(NoWarn);CS1573</NoWarn>
     <ClsCompliant>true</ClsCompliant>
-    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'uapaot'">true</IsPartialFacadeAssembly>
+    <GenrateTypeForwardsSourceCode Condition="'$(TargetGroup)' == 'uapaot'">true</GenrateTypeForwardsSourceCode>
```
Comment (safern) [inline src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: Typo on the property `GenrateTypeForwardsSourceCode` -> `GenerateTypeForwardsSourceCode`? Is this changing how we declare `PartialFacades`? Will this be the new property?

File: eng/referenceFromRuntime.targets [hunk 2]
```diff
@@ -58,12 +60,15 @@
...
-          Condition="'@(_referencePathFromRuntimeByFileName)' == '@(_remainingReferenceFromRuntimeWithNI)' AND '%(Identity)' != ''" />
+          Condition="'@(_referencePathFromRuntimeByFileName)' == '@(_remainingReferenceFromRuntimeWithNI)' AND '%(Identity)' != ''">
+        <Aliases>@(_remainingReferenceFromRuntimeWithNI->'%(Aliases)')</Aliases>
+      </_filteredReferencePathFromRuntimeByFileName>
 
       <_missingReferenceFromRuntime Include="@(_remainingReferenceFromRuntimeWithNI)" Exclude="@(_filteredReferencePathFromRuntimeByFileName)" />
 
       <!-- transform back to path -->
       <ReferencePath Include="@(_filteredReferencePathFromRuntimeByFileName->'%(ReferencePath)')" />
+      <ReferencePath Include="@(_filteredReferencePathFromRuntimeByFileName->'%(ReferencePath)')" Condition="'%(Aliases)' != ''" Aliases="" />
```
Comment (safern) [inline eng/referenceFromRuntime.targets]: @ericstj does it hurt anything on having a duplicate ReferencePath on an assembly?

Comment (ericstj) [reply eng/referenceFromRuntime.targets]: It's acceptable, but in some cases it could be a problem.  If someone specifically needed the alias to resolve a conflict in the global scope this wouldn't help them, but I am OK forcing them to alias both sides of that conflict.  I don't think we should make it too easy to write code against runtime assemblies that overlap with contract assemblies.  In those cases we should force the code to be explicit about which one it intends to use.

I think this at least warrants a comment though.

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj (hunk 1 above)
Comment (ericstj) [reply src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: I don't think we should be changing the property that triggers this

File: src/shims/manual/mscorlib.forwards.cs [hunk 3]
```diff
@@ -20,3 +20,5 @@
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.OrdinalComparer))]
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.UnitySerializationHolder))]
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Diagnostics.Contracts.ContractException))]
+// Genfacades tool is not able to generate a proper typeforward for System.void type.
```
Comment (ericstj) [inline src/shims/manual/mscorlib.forwards.cs]: Might be more helpful comment to indicate that this requires special C# syntax that we choose to handle here.

File: src/shims/manual/Directory.Build.props [hunk 4]
```diff
@@ -29,6 +29,10 @@
     <DefineConstants Condition="'$(TargetsNetCoreApp)' == 'true'">$(DefineConstants);netcoreapp</DefineConstants>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Reference Include="netstandard" />
```
Comment (ericstj) [inline src/shims/manual/Directory.Build.props]: I see why you did this.  I think an alternate fix would be to just modify the include statement below to include netstandard.dll in ReferencePath.  Either way has the same effect.

File: src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs [hunk 5]
```diff
@@ -8,12 +8,12 @@
 
 namespace System.Xml
 {
-    [Obsolete("This API supports the .NET Framework infrastructure and is not intended to be used directly from your code.", true)]
+    [Obsolete("This API supports the .NET Framework infrastructure and is not intended to be used directly from your code.", false)]
```
Comment (ericstj) [inline src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: Why are you changing these?

Comment (Anipik) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: If the second attribute is true, the compiler throws an error instead of throwing a warning

Comment (ericstj) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: Even with a pragma?   This needs to match https://github.com/dotnet/corefx/blob/16689aa0a933517dc45465fc6edb6223222414a6/src/System.Xml.ReaderWriter/ref/System.Xml.ReaderWriter.cs#L33.

Comment (Anipik) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: yes pragma just works on the warning not on errors.

Comment (ericstj) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: No we don't want to make that change in the ref.  My point was that we shouldn't change the behavior here.  It's actually less bad to change only in the implementation.  You may need to suppress an API compat message about the attributes being different.  I couldn't find a way to getaround the Obsolete(..., true) error: it appears that's a hard block.

Comment (Anipik) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: api compat is not checking attribute values as I am not getting any warning or error messages for this csproj

Comment (ericstj) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: Interesting that the compiler has this behavior with type of.  I can use the obsolete type in a method signature if the method itself is also marked obsolete.  There doesn't appear tobe a way to use the typeof() expression at all when the type is obsolete.  That's odd.

Comment (Anipik) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: yeah, what about if we create a class which implements ```IApplicationResourceStreamResolver``` and then use typeof(newclass).BaseType. I am not sure if baseType wil return the interface name

Comment (Anipik) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: we cant inherit the obsolete type

Comment (ericstj) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: You can't access a member of a type in a typeof expression: it has to be statically known at compile time.  @jaredpar are you aware of anyway to circumvent CS0619 when using typeof(foo) expression when foo is a error-obsoleted type?

Comment (jaredpar) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: @agocke may know a way here.

Comment (ericstj) [reply src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: FWIW I poked around the roslyn source and couldn't find a way to suppress this.  That's understandable considering its an *error* and a compiler with the ability to suppress errors would be pretty shaky.  I did find it interesting that the compiler would ignore the error so long as the caller was attributed with Obsolete, but we can't attribute a typeforward with Obsolete so... 🤷‍♂️ 

@Anipik I'm ok with you modifying implementation only to Obsolete(..., false) only for the type.

**Comment by Anipik** [comment]
cc @tannergooding

**Comment by ericstj** [comment]
Looks like there are some problems with the aliases in manual mscorlib shim.  Getting close.

**Comment by ericstj** [comment]
References for manual shims are here:
https://github.com/dotnet/corefx/blob/15680bb31d8aaa54b36d78d0132377d206435ba9/src/shims/manual/Directory.Build.props#L33-L35

None have aliases.  You'll need to add references with aliases.

**Comment by ericstj** [comment]
@Anipik and I looked at this.  It's happening because System.Private.CoreLib had InternalsVisibleTo mscorlib in 2.0, but the GenFacades source isn't considering InternalsVisibleTo when detecting seed preferences (nor did genfacades, but it didn't try to run typeforwards through the compiler).

To fix this he can flow assembly name into the task (simple name only to avoid cracking the SNK), then check the seed assembly for an InternalsVisibleTo attribute containing that assembly name.

**Comment by Anipik** [comment]
@ericstj i looked at the failure
```
D:\a\1\s\artifacts\obj\mscorlib\uap-Release\mscorlib.Forwards.cs(73,97): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'. [D:\a\1\s\src\shims\manual\mscorlib.csproj]
D:\a\1\s\artifacts\obj\mscorlib\uap-Release\mscorlib.Forwards.cs(73,12): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'. [D:\a\1\s\src\shims\manual\mscorlib.csproj]
    0 Warning(s)
```
The public key token for the system.private.corelib(one getting passed in reference paths) is ```b03f5f7f11d50a3a``` and it needs ```7cec85d7bea7798e```

any idea what should we do here ?

**Comment by ericstj** [comment]
7cec is the silverlight key. I wouldn’t expect that to show up at all.  You may want to see where that’s coming from by scanning the references of all assemblies being passed to the compiler.  I have a tool that does this and can help later tonight.

File: src/System.Memory/src/System.Memory.csproj [hunk 6]
```diff
@@ -5,6 +5,9 @@
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
     <Configurations>netcoreapp-Unix-Debug;netcoreapp-Unix-Release;netcoreapp-Windows_NT-Debug;netcoreapp-Windows_NT-Release;netcoreappaot-Windows_NT-Debug;netcoreappaot-Windows_NT-Release;uap-Windows_NT-Debug;uap-Windows_NT-Release</Configurations>
   </PropertyGroup>
+  <ItemGroup>
+    <GenFacadesIgnoreMissingType Include="System.Buffers.SequenceReader`1" />
```
Comment (danmosemsft) [inline src/System.Memory/src/System.Memory.csproj]: For the GenFacadesOmitType  and GenFacadesIgnoreMissingType  is a comment possible to explain "why" (in case you and @ericstj aren't available to ask)

File: src/shims/manual/mscorlib.csproj [hunk 7]
```diff
@@ -4,36 +4,83 @@
...
-    <SeedTypePreference Include="System.Runtime.InteropServices.WindowsRuntime.InterfaceImplementedInVersionAttribute" Assembly="System.Runtime.InteropServices.WindowsRuntime" />
-    <SeedTypePreference Include="System.Runtime.InteropServices.WindowsRuntime.ReadOnlyArrayAttribute" Assembly="System.Runtime.InteropServices.WindowsRuntime" />
-    <SeedTypePreference Include="System.Runtime.InteropServices.WindowsRuntime.ReturnValueNameAttribute" Assembly="System.Runtime.InteropServices.WindowsRuntime" />
-    <SeedTypePreference Include="System.Runtime.InteropServices.WindowsRuntime.WindowsRuntimeMarshal" Assembly="System.Runtime.InteropServices.WindowsRuntime" />
-    <SeedTypePreference Include="System.Runtime.InteropServices.WindowsRuntime.WriteOnlyArrayAttribute" Assembly="System.Runtime.InteropServices.WindowsRuntime" />
+    <Compile Include="mscorlib.forwards.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetsAot)' == 'true' or '$(TargetsUap)' == 'true'">
+    <ReferencePath Include="$(_RuntimePath)System.Reflection.Emit.dll" Aliases="System_Reflection_Emit" />
```
Comment (danmosemsft) [inline src/shims/manual/mscorlib.csproj]: Just curious, why the underscores?

File: src/System.Memory/src/System.Memory.csproj (hunk 6 above)
Comment (ericstj) [reply src/System.Memory/src/System.Memory.csproj]: Actually I don’t understand why we are suppressing this.

Comment (danmosemsft) [reply src/System.Memory/src/System.Memory.csproj]: OK, then in case @anipik isn't around to ask 😆

File: src/shims/manual/mscorlib.csproj (hunk 7 above)
Comment (ericstj) [reply src/shims/manual/mscorlib.csproj]: Maybe this should set RefPath instead. https://github.com/dotnet/corefx/blob/master/src/shims/manual/Directory.Build.props

File: src/System.Memory/src/System.Memory.csproj (hunk 6 above)
Comment (Anipik) [reply src/System.Memory/src/System.Memory.csproj]: Yeah i will add the comment.
We are using 2.0.0 version of Microsoft.CodeAnalysis.CSharp. This particular version has a bug where it is not able to parse ```public partial ref struct```  properly. It drops the public modifer.
We cant change the version of the assembly due to reason described here https://github.com/dotnet/arcade/pull/2221/files#diff-77a8f3b81eed042aa8a317f278b1037bR18

File: eng/Tools.props [hunk 8]
```diff
@@ -49,6 +49,7 @@
     <!-- analyzers -->
     <PackageReference Include="Desktop.Analyzers" Version="1.1.0" />
     <PackageReference Include="Microsoft.CodeAnalysis.Common" Version="2.9.0" />
+    <PackageReference Include="Microsoft.DotNet.CodeAnalysis" Version="$(MicrosoftDotNetCodeAnalysisPackageVersion)" />
```
Comment (ericstj) [inline eng/Tools.props]: What is the purpose of this change and why did you put it in the analyzers section 😕 

I'd expect all the dependencies of GenFacades to live next to the task itself.

File: src/System.Memory/src/System.Memory.csproj (hunk 6 above)
Comment (ericstj) [reply src/System.Memory/src/System.Memory.csproj]: Ah, now I remember.  We should file an issue on this and reference the issue from the comment.  It should go away once we can upgrade to the 3.0 CLI.

File: src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs [hunk 9]
```diff
@@ -8,12 +8,12 @@
...
     [EditorBrowsable(EditorBrowsableState.Never)]
     public interface IApplicationResourceStreamResolver
     {
         // Methods
-        [Obsolete("This API supports the .NET Framework infrastructure and is not intended to be used directly from your code.", true)]
+        [Obsolete("This API supports the .NET Framework infrastructure and is not intended to be used directly from your code.", false)]
```
Comment (ericstj) [inline src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: Remove this.  The member in the interface should not need to change.

File: src/System.Runtime/src/System.Runtime.csproj [hunk 10]
```diff
@@ -6,6 +6,9 @@
     <GenFacadesIgnoreMissingTypes Condition="'$(TargetsAOT)'=='true' OR '$(TargetGroup)' == 'uap'">true</GenFacadesIgnoreMissingTypes>
     <Configurations>netcoreapp-Unix-Debug;netcoreapp-Unix-Release;netcoreapp-Windows_NT-Debug;netcoreapp-Windows_NT-Release;netcoreappaot-Windows_NT-Debug;netcoreappaot-Windows_NT-Release;uap-Windows_NT-Debug;uap-Windows_NT-Release;uapaot-Windows_NT-Debug;uapaot-Windows_NT-Release</Configurations>
   </PropertyGroup>
+  <ItemGroup>
+    <GenFacadesOmitType Include="System.Void" />
```
Comment (ericstj) [inline src/System.Runtime/src/System.Runtime.csproj]: Add a comment here.  Same for other.

File: src/shims/manual/mscorlib.csproj [hunk 11]
```diff
@@ -4,36 +4,83 @@
...
+    <ReferencePath Include="$(_RuntimePath)System.Reflection.Emit.Lightweight.dll" Aliases="System_Reflection_Emit_Lightweight" />
+    <ReferencePath Include="$(_RuntimePath)System.Reflection.Emit.ILGeneration.dll" Aliases="System_Reflection_Emit_ILGeneration" />
+    <ReferencePath Include="$(_RuntimePath)System.Runtime.InteropServices.WindowsRuntime.dll" Aliases="System_Runtime_InteropServices_WindowsRuntime" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'">
+    <ReferencePath Include="$(_RuntimePath)Microsoft.Win32.Registry.dll" Aliases="Microsoft_Win32_Registry" />
```
Comment (ericstj) [inline src/shims/manual/mscorlib.csproj]: We should have a comment about this working around the fact that 2.0 s.p.c had internalsvisibleto mscorlib and included its own copy of these types.

**Comment by ericstj** [comment]
> The type 'System.Reflection.Emit.ParameterBuilder' is defined in multiple seed assemblies. If this is intentional, specify the alias for this type and project reference

This message should list the assemblies which contain the duplicate type.

**Comment by ericstj** [comment]
You shouldn't be emitting the source file when errors are encountered.  This is causing subsequent builds to skip the target which results in different errors upon incremental build.

File: src/shims/manual/mscorlib.csproj [hunk 12]
```diff
@@ -4,36 +4,83 @@
...
+    <Compile Include="mscorlib.forwards.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetsAot)' == 'true' or '$(TargetsUap)' == 'true'">
+    <ReferencePath Include="$(_RuntimePath)System.Reflection.Emit.dll" Aliases="System_Reflection_Emit" />
+    <ReferencePath Include="$(_RuntimePath)System.Reflection.Emit.Lightweight.dll" Aliases="System_Reflection_Emit_Lightweight" />
+    <ReferencePath Include="$(_RuntimePath)System.Reflection.Emit.ILGeneration.dll" Aliases="System_Reflection_Emit_ILGeneration" />
+    <ReferencePath Include="$(_RuntimePath)System.Runtime.InteropServices.WindowsRuntime.dll" Aliases="System_Runtime_InteropServices_WindowsRuntime" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.0'">
```
Comment (ericstj) [inline src/shims/manual/mscorlib.csproj]: This isn't working.  Use `'$(TargetGroup)' == 'netcoreapp2.0'`

**Comment by ericstj** [comment]
Ok, so I've found the issue with the UAPAOT build.  The problem here is that we have facades that are building against the coerclr System.Private.CoreLib and running against the .NETNative System.Private.CoreLib and the two have different keys. .NETNative corelib is b03f5f7f11d50a3a, CoreCLR is 7cec85d7bea7798e

The following are the problem projects:
```
System.Collections.Concurrent.dll
System.Diagnostics.TraceSource.dll
System.Globalization.Calendars.dll
System.Memory.dll
System.Text.Encoding.Extensions.dll
System.Threading.Tasks.Extensions.dll
```
We should add uap-aot configurations to these projects to fix this issue. 

@jkotas are you aware of this difference?  Seems like it could be something we'd want to fix for one.net.

**Comment by jkotas** [comment]
It is left over from days when ProjectN CoreLib was very different from CoreCLR CoreLib. IIRC, the different strong name was intentional so that it is harder to mix one and the other by accident. I agree that it would be nice to fix. The workaround you have suggested sounds fine to me.

File: src/shims/manual/mscorlib.csproj (hunk 11 above)
Comment (jkotas) [reply src/shims/manual/mscorlib.csproj]: Why do we still care about 2.0 s.p.c in current master?

File: src/System.Memory/src/System.Memory.csproj (hunk 6 above)
Comment (jkotas) [reply src/System.Memory/src/System.Memory.csproj]: Is this going to break folks dogfooding latest VS where msbuild loads newer Microsoft.CodeAnalysis.CSharp ? Should this rather launch in a separate process?

Comment (ericstj) [reply src/System.Memory/src/System.Memory.csproj]: @Anipik that's a good test case, can you try it out and make sure you don't have duplicate forwards?

Out of proc is possible but it feels pretty heavy handed just to workaround the MSBuild unification bug.

File: src/shims/manual/mscorlib.csproj (hunk 11 above)
Comment (ericstj) [reply src/shims/manual/mscorlib.csproj]: This is here because we are still adding new types in netstandard2.0-targeting assemblies that overlap with desktop surface area.  This builds a new set of compat shims that make use of those (assuming we weren't planning to deliver fixes already).  /cc @safern

**Comment by ericstj** [comment]
> The workaround you have suggested sounds fine to me.

Yeah, I almost want to write validation to catch this issue.  VerifyClosure ignores pkt today because the coeclr binder does, but maybe I should change that.

**Comment by Anipik** [comment]
> You shouldn't be emitting the source file when errors are encountered. This is causing subsequent builds to skip the target which results in different errors upon incremental build.

I am working on adding a new pr in arcade to add this as well all other features mentioned in the pr and discussed offline

File: src/System.IO.FileSystem/src/System.IO.FileSystem.csproj [hunk 13]
```diff
@@ -11,6 +11,7 @@
   <PropertyGroup Condition="'$(TargetsUnix)' == 'true'">
     <NoWarn>$(NoWarn);414</NoWarn>
   </PropertyGroup>
+  <!-- System.IO.Enumeration.FileSystemEntry is not correctly parsed by CsharpAnalyser so we are skipping this type from autogenerated typeforwards file.  -->
```
Comment (ericstj) [inline src/System.IO.FileSystem/src/System.IO.FileSystem.csproj]: nit: put this inside the itemgroup.

**Comment by ericstj** [comment]
Sounds good. Maybe we can work together next week to push this change in so we don't have to keep undoing the arcade insertions.

**Comment by Anipik** [comment]
> Sounds good. Maybe we can work together next week to push this change in so we don't have to keep undoing the arcade insertions. 

sure :)

File: src/shims/manual/mscorlib.csproj (hunk 11 above)
Comment (ericstj) [reply src/shims/manual/mscorlib.csproj]: I talked to @safern: we are only building these for netcoreapp2.0 and not delivering updates at all for netcoreapp2.1.  In that sense we can just delete this.  That will also save quite a bit of headache for @Anipik.

Comment (safern) [reply src/shims/manual/mscorlib.csproj]: https://github.com/dotnet/corefx/pull/36644

**Comment by Anipik** [comment]
@ericstj @safern 
when we  are targeting the build of the corefx against uapaot, the TargetGroup property is getting reassigned to ```uap``` and targetsAot is always false. is this a bug or am i missing something here ?

File: src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs [hunk 14]
```diff
@@ -8,7 +8,7 @@
 
 namespace System.Xml
 {
-    [Obsolete("This API supports the .NET Framework infrastructure and is not intended to be used directly from your code.", true)]
+    [Obsolete("This API supports the .NET Framework infrastructure and is not intended to be used directly from your code.", false)]
```
Comment (ericstj) [inline src/System.Private.Xml/src/System/Xml/IApplicationResourceStreamResolver.cs]: nit: add a comment here explaining the reason this differs from the reference assembly. EG: We must specify error=false in the implementation assembly so that we can typeforward to this type without hitting the compiler error.

File: src/shims/manual/mscorlib.csproj [hunk 15]
```diff
@@ -4,36 +4,36 @@
     <Configurations>Debug;Release</Configurations>
   </PropertyGroup>
   <ItemGroup>
-    <Compile Include="mscorlib.forwards.cs" />
+    <GenFacadesOmitType Include="System.Void" />
```
Comment (ericstj) [inline src/shims/manual/mscorlib.csproj]: nit:add comment.

**Comment by Anipik** [comment]
> LGTM. Thanks for pushing this feature through!

Thanks for the help :)

**Comment by ericstj** [comment]
> when we are targeting the build of the corefx against uapaot, the TargetGroup property is getting reassigned to uap and targetsAot is always false. is this a bug or am i missing something here ?

Not a bug.  It sounds like you are missing a configuration for `uapaot`.  In that case the config system will find closest compatible configuration which is `uap`.

File: src/shims/manual/Directory.Build.props [hunk 16]
```diff
@@ -2,8 +2,13 @@
   <PropertyGroup>
     <BuildConfigurations>
       netcoreapp;
-      netcoreappaot;
       uap;
+      uapaot;
+    </BuildConfigurations>
+    <!-- System.Private.CoreLib version is different in some projects for netcoreappaot builds. -->
```
Comment (ericstj) [inline src/shims/manual/Directory.Build.props]: Make sure we have an issue tracking the problems with manual shim builds.

Comment (Anipik) [reply src/shims/manual/Directory.Build.props]: yes I am just writing that one

Comment (Anipik) [reply src/shims/manual/Directory.Build.props]: https://github.com/dotnet/corefx/issues/36715
